src/Core/Form/IdentifiableObject/DataHandler/AttachmentFormDataHandler.php

```diff
@@ -0,0 +1,180 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataHandler;
+
+use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShop\PrestaShop\Core\Domain\Attachment\Command\CreateAttachmentCommand;
+use PrestaShop\PrestaShop\Core\Domain\Attachment\Command\EditAttachmentCommand;
+use PrestaShop\PrestaShop\Core\Domain\Attachment\ValueObject\AttachmentId;
+use PrestaShop\PrestaShop\Core\File\Uploader\FileUploaderInterface;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+/**
+ * Creates/updates attachment from data submitted in category form
+ */
+final class AttachmentFormDataHandler implements FormDataHandlerInterface
+{
+    /**
+     * @var CommandBusInterface
+     */
+    private $commandBus;
+
+    /**
+     * @var FileUploaderInterface
+     */
+    private $fileUploader;
+
+    /**
+     * @param CommandBusInterface $commandBus
+     * @param FileUploaderInterface $fileUploader
+     */
+    public function __construct(CommandBusInterface $commandBus, FileUploaderInterface $fileUploader)
+    {
+        $this->commandBus = $commandBus;
+        $this->fileUploader = $fileUploader;
+    }
+
+    /**
+     * @param array $data
+     *
+     * @return mixed|void
+     *
+     * @throws \PrestaShop\PrestaShop\Core\Domain\Attachment\Exception\AttachmentConstraintException
+     */
+    public function create(array $data)
+    {
+        do {
+            $uniqueFileName = sha1(microtime());
+        } while (file_exists(_PS_DOWNLOAD_DIR_ . $uniqueFileName));
```

Comment:
legacy code must not be used in `Core` name space. `_PS_DOWNLOAD_DIR_` is a legacy constant

Reply:
this logic should probably be moved to a separate service :thinking: